src/screens/tabs/Profile/modals/ActivityDetail/index.tsx

```diff
@@ -0,0 +1,114 @@
+import Clipboard from '@react-native-clipboard/clipboard';
+import { t } from 'i18next';
+import React, { RefObject } from 'react';
+import { View } from 'react-native';
+import { Modalize } from 'react-native-modalize';
+
+import {
+  ActionButton,
+  Header,
+  Modal,
+  Text,
+  Touchable,
+} from '@/components/common';
+import { Transaction, Wallet } from '@/interfaces/redux';
+import { useAppSelector } from '@/redux/hooks';
+import ActivityItem from '@/screens/tabs/components/ActivityItem';
+import { validateICNSName } from '@/utils/ids';
+import shortAddress from '@/utils/shortAddress';
+import { capitalize } from '@/utils/strings';
+
+import styles from './styles';
+
+interface Props {
+  modalRef: RefObject<Modalize>;
+  activity: Transaction;
+}
+
+interface RowProps {
+  title: string;
+  value: string;
+  onPress?: (address: string) => void;
+  showSelf?: boolean;
+  hideRow?: boolean;
+}
+
+const formatAddress = (address: string) =>
+  validateICNSName(address) ? address : shortAddress(address);
+
+const isMySelf = (address: string, currentWallet: Wallet) =>
```

Comment:
we could move formatAddress and isMySelf to utils/ids, what u think? and reuse it in other places

also, `isMySelf` -> `isOwnAddress`?

Reply:
Nice